### PR TITLE
PR frontend

### PR DESCRIPTION
**Pull request by Skema1114**

### CODE REVIEW

File: lib/data/network_service.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+class NetworkService {
+  final baseUrl = 'http://10.0.2.2:3000';
+
+  Future<List<dynamic>> fetchTodos() async {
+    try {
+      final response = await get(Uri.parse(baseUrl + '/todos'));
+      print(response.body);
+      return jsonDecode(response.body) as List;
+    } catch (e) {
+      print(e);
```
Comment (Yke222) [inline lib/data/network_service.dart]: We could use custom exceptions here.

File: lib/data/network_service.dart [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      return [];
+    }
+  }
+
+  Future<bool> patchTodo(Map<String, String> patchObj, int id) async {
+    try {
+      await patch(Uri.parse(baseUrl + '/todos/$id'), body: patchObj);
+      return true;
+    } catch (e) {
+      print(e);
```
Comment (Yke222) [inline lib/data/network_service.dart]: We could use custom exceptions here.

File: lib/data/network_service.dart [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      return false;
+    }
+  }
+
+  Future<Map> addTodo(Map<String, String> todoObj) async {
+    try {
+      final response = await post(Uri.parse(baseUrl + '/todos'), body: todoObj);
+      return jsonDecode(response.body);
+    } catch (e) {
+      print(e);
```
Comment (Yke222) [inline lib/data/network_service.dart]: We could use custom exceptions here.

File: lib/data/network_service.dart [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      return null;
+    }
+  }
+
+  Future<bool> deleteTodo(int id) async {
+    try {
+      await delete(Uri.parse(baseUrl + '/todos/$id'));
+      return true;
+    } catch (e) {
+      print(e);
```
Comment (Yke222) [inline lib/data/network_service.dart]: We could use custom exceptions here.

File: lib/presentation/screens/edit_todo_screen.dart [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          )
+        ],
+      ),
+    );
+  }
+
+  Widget _updateButton(context) {
+    return Container(
+      width: MediaQuery.of(context).size.width,
+      height: 50,
```
Comment (Yke222) [inline lib/presentation/screens/edit_todo_screen.dart]: Height could be responsive using MediaQuery or Responsive approach.

File: lib/presentation/screens/add_todo_screen.dart [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+            },
+            child: _addButton(context)),
+      ],
+    );
+  }
+
+  Widget _addButton(context) {
+    return Container(
+      width: MediaQuery.of(context).size.width,
+      height: 50,
```
Comment (Yke222) [inline lib/presentation/screens/add_todo_screen.dart]: Missing responsive height.

File: lib/presentation/screens/todos_screen.dart [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+              child: Padding(
+                padding: EdgeInsets.all(10),
+                child: Icon(Icons.add),
+              ),
+            )
+          ],
+        ),
+        body: BlocBuilder<TodosCubit, TodosState>(
+          builder: (context, state) {
+            if (!(state is TodosLoaded)) {
```
Comment (Yke222) [inline lib/presentation/screens/todos_screen.dart]: if state != TodosLoaded

File: README.md [hunk 8]
```diff
@@ -1,81 +1,31 @@
...
+- Listagem de Produtos;
+- Remoção de Produtos (mas não está atualizando a tela na sequência);
+- Remoção de Lojas (também não atualiza a tela após deletar);
+- Persistência de dados na memória interna;
+- Persistência de dados do usuário para que o mesmo mantenha-se logado;
+- A estrutura foi baseada em um MVC mas utiliza o básico de gerenciamento de estados utilizando o gerenciador nativo do Flutter (setState).
+
+# Dificuldades encontradas
+
+A maior dificuldade nesse desenvolvimento (e o que gastou um bom pouco do tempo) foi a tentativa da implementação do BLoC. o mesmo não deu certo, então parti para o BLoC/Cubit, que também, ficou com problemas na realização da chamada POST da API, o GET estava funcionando bem mas o POST não funcionou. Nesse tempo eu parti pro React Native, comecei a desenvolver e fiz o CRUD da Loja, só que como o desafio é voltado ao Flutter, resolvi voltar e continuar no desenvolvimento, com isso utilizei meio que o padrão MVC, não é o mais recomendado para utilizar, mas visto a quantidade de tempo que ainda restavam.
```
Comment (joaomantovani) [inline README.md:31]: Nice README :rocket:

File: lib/controllers/auth_controller.dart [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+      if (res.statusCode == 200) {
+        return true;
+      } else {
+        return false;
+      }
+    } catch (e) {
+      print(e);
+      return false;
+    }
+  }
```
Comment (joaomantovani) [inline lib/controllers/auth_controller.dart:28]: It's a good practice to create single services for each responsability :+1:

**Review by joaomantovani — APPROVED** [review]
The code is well structured, use in most files the single responsability approach, the widgets and variables names are easy to understand.

**Comment by Yke222** [comment]
Added suggestions to provide a better reading and code improvements

**Review by Yke222 — APPROVED** [review]

**Pull request closed without merging**